androidplot-core/src/main/java/com/androidplot/xy/PanZoom.java

```diff
@@ -347,7 +378,14 @@ protected void calculateZoom(RectF newRect, float scale, boolean isHorizontal) {
                 }
             }
         } else {
-            final RectRegion limits = plot.getOuterLimits();
+            // zoom limited and increment by value StepMode?
+            if (zoomLimit == ZoomLimit.MIN_TICKS && (plot.getRangeStepMode() == StepMode.INCREMENT_BY_FIT || plot.getRangeStepMode() == StepMode.INCREMENT_BY_VAL)) {
```

Comment:
The `getRangeStepMode()` can be dropped. Then we will call `getDomainStepValue()` also in `StepMode.SUBDIVIDE` or `StepMode.INCREMENT_BY_PIXELS`. I guess that is a little faster and easier.